gui/src/controls/control.ts

```diff
@@ -768,6 +768,13 @@ module BABYLON.GUI {
         
             this._fontOffset = Control._GetFontOffset(this._font);
         }
+        
+        public measureForIdeal() {
+            this._measure();
+            
+            this.width = this._currentMeasure.width + 'px';

```

Comment:
Did you test it?

I don't think it will work as this.width will be reevaluated again and then scale down or up and then this._currentMeasure will be changed accordingly.